xdsl/transforms/canonicalization_patterns/arith.py

```diff
@@ -189,6 +189,41 @@ def match_and_rewrite(self, op: arith.SelectOp, rewriter: PatternRewriter):
             rewriter.replace_matched_op((), (op.lhs,))
 
 
+class CmpfOpFoldSelectPattern(RewritePattern):
+    """
+    %1 = arith.cmpf  ogt, %0, %cst : f64
+    %2 = arith.select %1, %0, %cst : f64
+    """
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: arith.CmpfOp, rewriter: PatternRewriter):
+        if len(op.result.uses) != 1:
```

Comment:
Is it not better to continue anyway in this case?

Reply:
Hmm not sure, it's a neat way of getting the select I'm interested in. Maybe this should be a canonicalisation pattern on the select op in the first place? It's just that the logic is rather cmpf-specific